### PR TITLE
Gcw 2145 appointment feature

### PR DESCRIPTION
**Pull request by namrataukirde**
Hi Team,

This PR adds flow for booking appointment.

Please review.

Thanks

### CODE REVIEW

File: app/components/add-request.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  order: null,
+
+  packageTypeName: Ember.computed('request.packageType.name', function(){
+    return this.get('request.packageType.name');
+  }),
+
+  actions: {
+    deleteRequest(reqId) {
+      this.get("messageBox").custom(
+        "Remove this request from order " + this.get("model.code"),
```
Comment (Darshan-Chauhan) [inline app/components/add-request.js]: i18n?

File: app/components/focus-textfield.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import Ember from "ember";
+
+export default Ember.TextField.extend({
+  tagName: "input",
+  type:    "text",
+  maxlength: "25",
+  attributeBindings: [ "name", "id", "value", 'placeholder'],
+  cordova: Ember.inject.service(),
```
Comment (Darshan-Chauhan) [inline app/components/focus-textfield.js]: Do we need this service here?

File: app/components/focus-textfield.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  store: Ember.inject.service(),
```
Comment (Darshan-Chauhan) [inline app/components/focus-textfield.js]: ^

File: app/controllers/order/goods_details.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import AjaxPromise from 'browse/utils/ajax-promise';
+
+export default Ember.Controller.extend({
+  queryParams: ["typeId", "fromClientInformation"],
+  order: Ember.computed.alias("model"),
+  typeId: null,
+  fromClientInformation: false,
+  qty: null,
+  otherDetails: "",
+  requestCount: 3,
```
Comment (Darshan-Chauhan) [inline app/controllers/order/goods_details.js]: Why is this set to 3?

File: app/controllers/order/search_code.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
+import Ember from 'ember';
+import { translationMacro as t } from "ember-i18n";
+import AjaxPromise from 'browse/utils/ajax-promise';
+
+const { getOwner } = Ember;
+
+export default Ember.Controller.extend({
```
Comment (Darshan-Chauhan) [inline app/controllers/order/search_code.js]: I think we can avoid lot of lines if we extend this controller from `search_module`.
What do you think??

File: app/controllers/request_purpose.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  selectedId: null,
+  isSelfSelected: Ember.computed.equal("selectedId", "organisation"),
+  user: Ember.computed.alias('session.currentUser'),
+
+  actions: {
+    clearDescription() {
+      this.set("description", "");
+    },
+    
+    createOrderWithRequestPuropose(){
```
Comment (Darshan-Chauhan) [inline app/controllers/request_purpose.js]: Can we extract some functionality in another function?

File: app/routes/order/client_information.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import AuthorizeRoute from './../authorize';
+
+export default AuthorizeRoute.extend({
+  model() {
+    var orderId = this.paramsFor('order').order_id;
+    var order = this.store.peekRecord('order', orderId) || this.store.findRecord('order', orderId);
+
+    return Ember.RSVP.hash({
+      order: order,
+      beneficiary: order.get('beneficiary')
```
Comment (Darshan-Chauhan) [inline app/routes/order/client_information.js]: Is this working?? Is it waiting for Order request to get completed?

File: app/routes/order/confirm_booking.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import AuthorizeRoute from './../authorize';
+
+export default AuthorizeRoute.extend({
+	model() {
```
Comment (Darshan-Chauhan) [inline app/routes/order/confirm_booking.js]: Indentation

**Comment by namrataukirde** [comment]
Yes @Darshan-Chauhan , I am moving plain english to translation files. :+1:

File: app/controllers/order/goods_details.js (hunk 4 above)
Comment (namrataukirde) [reply app/controllers/order/goods_details.js]: This is not needed. I will remove it. Thanks 👍

File: app/controllers/order/search_code.js (hunk 5 above)
Comment (namrataukirde) [reply app/controllers/order/search_code.js]: Okay, Let me check. Thanks for flagging this. :)

File: app/controllers/request_purpose.js (hunk 6 above)
Comment (namrataukirde) [reply app/controllers/request_purpose.js]: Yup Sure. Thanks for input. 👍

File: app/controllers/order/search_code.js (hunk 5 above)
Comment (namrataukirde) [reply app/controllers/order/search_code.js]: Good suggestion but Currently we do not have search_module in browse to extend from it. We have that in stock so we need to move that to goodcity-lib first then we can use that for both stock and browse.

**Pull request merged**